### PR TITLE
fix-ref/sha/tag

### LINKED ISSUE #106: getting "failed to build bootstrap manifests" since v0.2.5
since the [ref/sha/tag](https://github.com/argoproj-labs/argocd-autopilot/pull/98) change we are getting this error when running the `repo bootstrap` command:
```
FATAL failed to build bootstrap manifests: accumulating resources: accumulation err='accumulating resources from 'github.com/argoproj-labs/argocd-autopilot/manifests?tag=v0.2.5': evalsymlink failure on '/auto-pilot518561277/github.com/argoproj-labs/argocd-autopilot/manifests?tag=v0.2.5' : lstat /auto-pilot518561277/github.com: no such file or directory': evalsymlink failure on '/private/var/folders/s7/ydmj1nh917ncbk9y1mhn2dfm0000gn/T/kustomize-480526136/manifests?tag=v0.2.5' : lstat /private/var/folders/s7/ydmj1nh917ncbk9y1mhn2dfm0000gn/T/kustomize-480526136/manifests?tag=v0.2.5: no such file or directory
```

Currently I am able to get passed it by providing the `--app` flag with the correct kustomize URL, like so:
`argocd-autopilot repo bootstrap --app "github.com/argoproj-labs/argocd-autopilot/manifests?ref=v0.2.5"`

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: cmd/commands/repo.go [hunk 1]
```diff
@@ -165,7 +165,7 @@ func NewRepoBootstrapCommand() *cobra.Command {
 
 	<BIN> repo bootstrap --repo https://github.com/example/repo/path/to/installation_root
 `),
-		PreRun: func(_ *cobra.Command, _ []string) { cloneOpts.Parse() },
+		PreRun: func(cmd *cobra.Command, args []string) { cloneOpts.Parse() },
```
Comment (roi-codefresh) [inline cmd/commands/repo.go:168]: why this change?

File: pkg/git/repository.go [hunk 2]
```diff
@@ -160,7 +159,7 @@ func (o *CloneOptions) URL() string {
 }
 
 func (o *CloneOptions) Revision() string {
-	return plumbing.ReferenceName(o.revision).Short()
+	return o.revision
```
Comment (roi-codefresh) [inline pkg/git/repository.go:162]: do we even need this?

File: pkg/git/repository.go [hunk 3]
```diff
@@ -244,6 +243,39 @@ var initRepo = func(ctx context.Context, opts *CloneOptions) (*repo, error) {
...
+		remotes, err := r.Remotes()
+		if err != nil {
+			return err
+		}
+
+		if len(remotes) == 0 {
+			return ErrNoRemotes
+		}
+
+		remoteref := fmt.Sprintf("%s/%s", remotes[0].Config().Name, ref)
```
Comment (roi-codefresh) [inline pkg/git/repository.go:262]: lets add some debug msgs here and before the checkout

**Review by roi-codefresh — APPROVED** [review]

File: cmd/commands/repo.go (hunk 1 above)
Comment (noam-codefresh) [reply cmd/commands/repo.go:168]: It was probably caused by the revert. I'll revert the revert

**Review by roi-codefresh — APPROVED** [review]

File: pkg/git/repository.go (hunk 2 above)
Comment (noam-codefresh) [reply pkg/git/repository.go:162]: Yeah, when creating an Argo CD Application, the Source.Revision field will contain this value.

**Pull request merged**